src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Progress } from "@/components/ui/progress";
+import { api } from "~/utils/api";
+
+const ProgressBar = ({ bookId, chapterId,sectionId }: { bookId: string; chapterId: string; sectionId: string }) => {
+  const { data: retrievedBook } = api.book.getBook.useQuery(bookId);
+
+  if (!retrievedBook) return <p>No book found!</p>;
+
+  const currentChapterIndex = retrievedBook.chapters.findIndex(chapter => chapter.id === chapterId);
+  const currentChapterTitle = retrievedBook.chapters[currentChapterIndex]?.title ?? 'Unknown Chapter';
+  const currentChapter = retrievedBook.chapters[currentChapterIndex];
```

Comment:
Switch the order of this line and the line above

Reply:
Saves a check

Reply:
const currentChapterIndex = retrievedBook.chapters.findIndex((chapter) => chapter.id === chapterId,);
const currentChapter = retrievedBook.chapters[currentChapterIndex];
const currentChapterTitle = currentChapter?.title;
  
Like this?

Reply:
Yeah